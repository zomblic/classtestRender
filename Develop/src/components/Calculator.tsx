import { useState } from "react";

interface CalculatorState {
  total: number;
  previousTotal: number;
}

export default function Calculator() {
  const [calcState, setCalcState] = useState<CalculatorState>({ total: 0, previousTotal: 0 });
  const [value, setValue] = useState(0);

  const handleAddClick = () => {
    setCalcState({ total: calcState.total + value, previousTotal: calcState.total});
  };

  return (
    <div className="calculator">
      <h2>Simple Calculator</h2>
      <p>
        <input
          type="text"
          id="value"
          name="value"
          onChange={(event) => { setValue(Number(event.target.value)); }}
          value={value}
        />
      </p>
      <button className="btn btn-primary" type="button" onClick={handleAddClick}> Add </button>
      <p className="card-text">Total: {calcState.total}</p>
      <p className="card-text">Previous Total: {calcState.previousTotal}</p>
    </div>
  );
}
