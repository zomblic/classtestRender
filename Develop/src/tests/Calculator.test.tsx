import pretty from 'pretty';
import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  it('should contain the expected text', () => {
    render(<Calculator />);

    const itemElement = document.querySelector('h2');

    if (itemElement) {
      expect(itemElement.textContent).toBe('Simple Calculator');  
    }
    
  });

  it('should match snapshot', () => {
    render(<Calculator />);

    const calculatorElement = document.querySelector('.calculator');

    if (calculatorElement) {
      expect(pretty(calculatorElement.innerHTML)).toMatchSnapshot();
    }
  });
});
