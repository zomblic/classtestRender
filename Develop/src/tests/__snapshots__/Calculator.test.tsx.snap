// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Calculator > should match snapshot 1`] = `
"<h2>Simple Calculator</h2>
<p><input type="text" id="value" value="0" name="value"></p><button class="btn btn-primary" type="button"> Add </button>
<p class="card-text">Total: 0</p>
<p class="card-text">Previous Total: 0</p>"
`;
